src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/JCloudsProvider.java

```diff
@@ -124,14 +126,16 @@ public void deleteInstance(Infrastructure infrastructure, String instanceId) {
     public List<ScriptResult> executeScriptOnInstanceId(Infrastructure infrastructure, String instanceId,
             InstanceScript instanceScript) {
         ExecResponse execResponse;
-
+        RunScriptOptions runScriptOptions = null;
         try {
+            String scriptToExecuteString = buildScriptToExecuteString(instanceScript);
+            runScriptOptions = buildScriptOptionsWithInstanceId(instanceScript, instanceId, infrastructure);
             execResponse = getComputeServiceFromInfastructure(infrastructure).runScriptOnNode(instanceId,
-                                                                                              buildScriptToExecuteString(instanceScript),
-                                                                                              buildScriptOptionsWithInstanceId(instanceScript,
-                                                                                                                               instanceId,
-                                                                                                                               infrastructure));
+                                                                                              scriptToExecuteString,
+                                                                                              runScriptOptions);
         } catch (Exception e) {
+            log.error("Script cannot be run on instance with id: " + instanceId + ". RunScriptOptions=" +
+                      runScriptOptions, e);
             throw new RuntimeException("Script cannot be run on instance with id: " + instanceId, e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)